src/dex-ui/pages/CreateProposal/ProposalForms/ContractUpgradeProposalForm.tsx

```diff
@@ -1,16 +1,127 @@
-import { FormControl, Input } from "@chakra-ui/react";
+import { FormControl, Input, FormErrorMessage, Flex, Button, Spacer, Text } from "@chakra-ui/react";
+import { ReactElement, useState } from "react";
+import { useForm } from "react-hook-form";
+import { useNavigate } from "react-router-dom";
+import { TextEditor } from "../../../../dex-ui-components";
+import { ProposalType } from "../../../store/governanceSlice";
+import { FileUploader, LoadingDialog } from "../../../../dex-ui-components";
+import { useDeployContract } from "../../../hooks/governance/useContractDeploy";
+
+interface ContractUpgradeProposalFormData {
+  title: string;
+  description: string;
+  linkToDiscussion: string;
+  abiFile: string;
+  abiFileName: string;
+  newImplementationAddress: string;
+  proxyAddress: string;
+}
+
+export function ContractUpgradeProposalForm(): ReactElement {
+  const navigate = useNavigate();
+  const {
+    handleSubmit,
+    register,
+    setValue,
+    formState: { errors, isSubmitting },
+  } = useForm<ContractUpgradeProposalFormData>();
+
+  const handleCancelClick = () => navigate("/governance");
+  const deployContract = useDeployContract();
+  const [fileName, setFileName] = useState("");
+  if (deployContract.data?.id) {
+    setValue("newImplementationAddress", deployContract.data?.id);
+    setValue("proxyAddress", deployContract.data?.id);
+  }
+  const onFileReadAction = (file: string, fileName: string) => {
+    setFileName(fileName);
+    deployContract.mutate({ abiFile: file });
```

Comment:
To Submit the Proposal we need the new contract id and proxy address (new implementation address in UI) which can be retrieved only by deploying the contract first, we also need transparent proxy address to publish the form from where it will come to the user that we need to check with team. once proposal in published and executed after successful voting then the contract will be upgraded.